src/VirtualTree.lua

```diff
@@ -0,0 +1,33 @@
+local ShallowWrapper = require(script.Parent.ShallowWrapper)
+local Type = require(script.Parent.Type)
+
+local config = require(script.Parent.GlobalConfig).get()
+
+local VirtualTree = {}
+
+function VirtualTree.new(rootNode, internalDataSymbol)
+	local tree = {
+		[Type] = Type.VirtualTree,
+		[internalDataSymbol] = {
+			rootNode = rootNode,
+			mounted = true,
```

Comment:
we should pass mounted as a parameter instead of assuming a tree is always created as mounted (though that's always true right now)